src/libexec/zmdiaglog

```diff
@@ -121,12 +121,11 @@ sub save_heap_info($$$$) {
 
     $histo_cmd = "$JMAP -histo:live $pid";
     $dump_file = "$destination/heapdump-live.hprof";
-    $dump_cmd  = "$SU '$JMAP -dump:live,file=$dump_file $pid'";
-
+    $dump_cmd  = " '$JMAP -dump:live,file=$dump_file $pid'";
     logmsg "Retrieving JVM $version heap histogram\n";
-    exec_with_timeout( "$SU '$histo_cmd' > heap.histo 2>&1", $timeout );
+    exec_with_timeout( " '$histo_cmd' > heap.histo 2>&1", $timeout );
     logmsg "Saving JVM $version heapdump\n";
-    my $success = exec_with_timeout( "$dump_cmd", $timeout );
+    my $success =  system("bash -c $dump_cmd "); 
```

Comment:
No timeout anymore?

Reply:
No we cannot use the timeout process anymore, because that forks the process as zimbra. And for heapdump cannot be run as zimbra user, but as root user.